File: trunk/src/dotnet/yaxgl/YaxglLibrary/Container/WindowManager.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.Windows.Forms;
using System.Xml;

namespace de.yaxgl
{
    public class WindowManager
    {
        IList<Window> windows = null;
        private static WindowManager instance = null;
        private static bool initialized = false;

        
        class NotJetInitilaizedException : Exception 
        { 
            public NotJetInitilaizedException()
            {
            
            }
            public NotJetInitilaizedException(string message)
                : base(message)
            {
            
            }
        }

        private WindowManager()
        {
                
            windows = new List<Window>();
            Application.EnableVisualStyles();
            Application.SetCompatibleTextRenderingDefault(false);
        }


        public static void initialize(LookAndFeel lookAndFeel)
        {
            if (lookAndFeel != LookAndFeel.System)
                throw new NotSupportedException("Error: The LookAndFeel." + lookAndFeel.ToString() + " style is not supported in C# .NET applications");
            
            initialized=true;
        }
        
        
        public static WindowManager getInstance()
        {
            if (!initialized)
                throw new NotJetInitilaizedException("Error: You have to initialize the WindowManager with a LookAndFeel before you can get an instance");

            if (instance == null)
            {
                instance = new WindowManager();
            }
            return instance;
        }
        
        public Window createWindow(string xmlfile,Object EventReciever)
        {
            Window window = new Window(xmlfile);
            window.registerEventHandlers(EventReciever);
            windows.Add(window);
            return window;
        }


        /* this method will not return until closing the window
         * starts a new application by given Window baseWindow
         * */
        public void run(Window baseWindow)
        {
            if (baseWindow == null)
                throw new NullReferenceException("Error: cannot run application");

            Application.Run((Form)baseWindow.getNativeComponent());
        }
        
        public void show(Window window)
        {
            if(window!=null)
                window.show();            
        }

        public void showModal(Window window, Window parentWindow)
        {
            if(window!=null)
                window.showDialog(parentWindow);
        }

        public void hide(Window window)
        {
            if (window != null)
                window.hide();
        }

        public void close(Window window)
        {
            if (window != null)
            {
                if(windows.Contains(window))
                {
                    windows.Remove(window);
                }
                window.close();
            }
        }

        public IList<Window> getWindowsByID(string ID)
        {
            IList<Window> matchingWindows = new List<Window>();
            foreach (Window window in windows)
            {
                if (window.getID().Equals(ID))
                    matchingWindows.Add(window);
            
            }
            return matchingWindows;
        }
    }
}

File: src/java/YaxglLibrary/de/yaxgl/Container/WindowManager.java
package de.yaxgl.Container;

import java.util.ArrayList;
import java.util.List;

import org.eclipse.swt.widgets.Display;
import org.eclipse.swt.widgets.Shell;

import de.yaxgl.Helper.LookAndFeel;

public class WindowManager {

	 List<Window> windows = null;
     private static WindowManager instance = null;
     private static boolean initialized = false;
     private static Display display=null;
     
     private WindowManager()
     {
             
         windows = new ArrayList<Window>();
     }


     public static void initialize(LookAndFeel lookAndFeel) throws Exception
     {
         if (lookAndFeel != LookAndFeel.System)
        	 throw new Exception("The selected LookAndFeel is not supported.");
        	 
         initialized=true;
     }
     
     
     public static WindowManager getInstance() throws Exception
     {
         if (!initialized)
             throw new Exception("The WindowManager must be initialized before you can have an instance.");
         
         if (instance == null)
         {
             display=new Display();
        	 instance = new WindowManager();
         }
         return instance;
     }
     
     public Window createWindow(String xmlfile,Object EventReciever)
     {
         Window window = new Window(display,xmlfile);
         window.registerEventHandlers(EventReciever);
         windows.add(window);
         return window;
     }


     /* this method will not return until closing the window
      * starts a new application by given Window baseWindow
      * */
     public void run(Window baseWindow)
     {
    	 Shell shell=null;
    	 if (baseWindow != null)
    	 {
    		 shell=(org.eclipse.swt.widgets.Shell)baseWindow.getNativeComponent();
    		 baseWindow.show();
    	 }
            
    	 while (!shell.isDisposed ()) {
  			if (!display.readAndDispatch ()) display.sleep ();
  		}
  		display.dispose ();
     }
     
     public void show(Window window)
     {
         if(window!=null)
             window.show();            
     }

     public void showModal(Window window, Window parentWindow)
     {
         if(window!=null)
             window.showDialog(parentWindow);
     }

     public void hide(Window window)
     {
         if (window != null)
             window.hide();
     }

     
     public void close(Window window)
     {
         if (window != null)
         {
             if(windows.contains(window))
             {
                 windows.remove(window);
             }
             window.close();
         }
     }

     public List<Window> getWindowsByID(String ID)
     {
         List<Window> matchingWindows = new ArrayList<Window>();
         for(Window window : windows)
         {
             if (window.getID().equals(ID))
                 matchingWindows.add(window);
         }
         return matchingWindows;
     }

}
